services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/dataset.rb

```diff
@@ -37,20 +37,24 @@ class Dataset < ApplicationRecord
 
         delegate :stem_and_conjunction, to: :stem_vault
 
+        scope :whole, -> { where(parent_id: nil) }
+
         attr_accessor :file
 
         before_validation :set_version
 
-        def dataslices = where(parent_id: id)
+        def data_subsets = self.class.where(parent_id: id)
```

Comment:
Take or leave, but you can define this as a `has_many` relationship (with a corresponding `belongs_to`).

Reply:
Good point